### PR TITLE
feat(eda.plot and eda.plot_missing): now can specify column types

### PR DESCRIPTION
**Pull request by dylanzxc**
# Description

plot() and plot_missing() now can generate graphs according to the data
types specified by users.

# How Has This Been Tested?

Two unit tests have been created for plot() and plot_missing().

# Snapshots:

Include snapshots for easier review.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have already squashed the commits and make the commit message conform to the project standard.
- [X] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jinglinpeng** [comment]
please take a look at the circleci, and squash the commits.

**Comment by dylanzxc** [comment]
Yes I'm fixing those errors now. Shall I squash Weiyuan's commit? 
I can't run just ci on windows, so I can only pr first and resolve those bugs.

File: dataprep/eda/basic/render.py [hunk 1]
```diff
@@ -1176,11 +1176,11 @@ def render_basic(
                 False,
             )
             figs.append(fig)
-        elif dtype == DType.Numerical:
+        elif is_dtype(dtype, Continuous()):
             df, miss_pct = data
             fig = hist_viz(df, miss_pct, col, yscale, plot_width, plot_height, False)
             figs.append(fig)
-        elif dtype == DType.DateTime:
+        elif dtype == DateTime:
```
Comment (jinglinpeng) [inline dataprep/eda/basic/render.py:1183]: Should use `is_dtype(dtype, DateTime())`?

File: dataprep/eda/dtypes.py [hunk 2]
```diff
@@ -20,22 +24,152 @@
...
+    """
+    Type Text, Subtype of Nominal
+    """
+
+
+############## End of the Type Tree ##############
+
+
+def detect_dtype(
+    col: dd.Series, known_dtypes: Optional[Dict[str, Union[DType, str]]] = None
```
Comment (jinglinpeng) [inline dataprep/eda/dtypes.py:89]: Please explain the meaning of each parameter in the docstring. (follow https://github.com/sfu-db/dataprep/wiki/Docstring-Best-Practice)

File: dataprep/eda/dtypes.py [hunk 3]
```diff
@@ -20,22 +24,152 @@
...
+) -> DType:
+    """
+    Given a column, return its type
+    """
+    if known_dtypes:
+        for col_name, dtype in known_dtypes.items():
```
Comment (jinglinpeng) [inline dataprep/eda/dtypes.py:95]: Seems the for loop is not necessary. Could we write `if col_name in known_dtypes` instead?

File: dataprep/eda/dtypes.py [hunk 4]
```diff
@@ -20,22 +24,152 @@
...
+        "Ordinal": Ordinal,
+        "Nominal": Nominal,
+        "Numerical": Numerical,
+        "Continuous": Continuous,
+        "Discrete": Discrete,
+        "DateTime": DateTime,
+        "Text": Text,
+    }
+
+    for str_dtype, dtype in str_dic.items():
```
Comment (jinglinpeng) [inline dataprep/eda/dtypes.py:154]: Should we map categorical to nominal, and numerical to continuous in this function? If user input `plot(df, a = Categorical)`, will it be handled correctly? (it should be transformed to nominal somewhere).

File: dataprep/eda/basic/__init__.py [hunk 5]
```diff
@@ -31,6 +32,7 @@ def plot(
     value_range: Optional[Tuple[float, float]] = None,
     yscale: str = "linear",
     tile_size: Optional[float] = None,
+    dtypes: Optional[Dict[str, Union[DType, str]]] = None,
```
[the remark this answers is not in the document]
Comment (dylanzxc) [reply dataprep/eda/basic/__init__.py:35]: okay, I will change dtypes to `dtype: Optional[Union[Dict[str, Union[Dtype, str]], DType]]` and add another condition for dtype = single DType to transform all columns' dtypes.

File: dataprep/eda/basic/render.py (hunk 1 above)
Comment (dylanzxc) [reply dataprep/eda/basic/render.py:1183]: Yeah, you are right, I will fix these.

File: dataprep/eda/dtypes.py (hunk 2 above)
Comment (dylanzxc) [reply dataprep/eda/dtypes.py:89]: Okay, will explain.

File: dataprep/eda/dtypes.py (hunk 4 above)
Comment (dylanzxc) [reply dataprep/eda/dtypes.py:154]: These two mappings have been put under detect_types line97-107
```
               if (
                    isinstance(normalize_dtype(dtype), Categorical) is True
                    and isinstance(normalize_dtype(dtype), Ordinal) is False
                    and isinstance(normalize_dtype(dtype), Nominal) is False
                ):
                    return Nominal()
                elif (
                    isinstance(normalize_dtype(dtype), Numerical) is True
                    and isinstance(normalize_dtype(dtype), Continuous) is False
                    and isinstance(normalize_dtype(dtype), Discrete) is False
                ):
                    return Continuous()
```
I will optimize this code. But as nomalize_dtype will return instance such as Categorical(), it's not straightforward to map, I combined three isinstance(). Please let me know if there is a better way.

Comment (jinglinpeng) [reply dataprep/eda/dtypes.py:154]: That works for me.

File: dataprep/eda/basic/__init__.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dylanzxc) [reply dataprep/eda/basic/__init__.py:35]: image
image

File: dataprep/eda/basic/compute.py [hunk 6]
```diff
@@ -48,6 +60,7 @@ def compute(
     agg: str = "mean",
     sample_size: int = 1000,
     value_range: Optional[Tuple[float, float]] = None,
+    dtype: Optional[Union[Dict[str, Union[DType, str]], DType]] = None,
```
[the remark this answers is not in the document]
Comment (dovahcrow) [reply dataprep/eda/basic/compute.py:63]: Also, this type signature does not capture the `by-class` case. Please read `typing.Type` to make `by-class` work.

File: dataprep/eda/basic/compute.py [hunk 7]
```diff
@@ -138,23 +171,42 @@ def compute_overview(
...
+        Specify Data Types for designated column or all columns.
+        E.g.  dtype = {"a": Continuous, "b": "Nominal"} or
+        dtype = {"a": Continuous(), "b": "nominal"}
+        or dtype = Continuous() or dtype = "Continuous" or dtype = Continuous()
     """
 
     datas: List[Any] = []
-    col_names_dtypes: List[Tuple[str, DType]] = []
+    col_names_dtypes: List[
+        Tuple[
```
[the remark this answers is not in the document]
Comment (dylanzxc) [reply dataprep/eda/basic/compute.py:183]: Yeah sure, I forgot to delete that part after we decided to change to by instance representation.

File: dataprep/eda/basic/compute.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dylanzxc) [reply dataprep/eda/basic/compute.py:63]: We only use `DTypeDef = Union[Dict[str, Union[DType, str]], DType]` in all modules.
In this case,  is it necessary?

File: dataprep/eda/basic/compute.py [hunk 8]
```diff
@@ -87,37 +100,57 @@ def compute(
     value_range: Optional[Tuple[float, float]], default None
         The lower and upper bounds on the range of a numerical column.
         Applies when column x is specified and column y is unspecified.
+    dtype: Optional[Union[Dict[str, Union[DType, str]], DType]], default None
```
[the remark this answers is not in the document]
Comment (dylanzxc) [reply dataprep/eda/basic/compute.py:103]: okay

File: dataprep/eda/basic/compute.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (dovahcrow) [reply dataprep/eda/basic/compute.py:63]: I think it's ok to define more to make things clear.

File: dataprep/eda/dtypes.py [hunk 9]
```diff
@@ -20,22 +24,177 @@
...
+    ----------
+    col: dask.datafram.Series
+        A dataframe column
+    known_dtype: Optional[Union[Dict[str, Union[DType, str]], DType]], default None
+        A dictionary or single DType given by users to specify the types for designated columns or
+        all columns. E.g.  known_dtype = {"a": Continuous, "b": "Nominal"} or
+        known_dtype = {"a": Continuous(), "b": "nominal"} or
+        known_dtype = Continuous() or known_dtype = "Continuous" or known_dtype = Continuous()
     """
+    if known_dtype:
```
Comment (dovahcrow) [inline dataprep/eda/dtypes.py:104]: Maybe it is better to make it early return here.

**Pull request closed without merging**